api/s2n.h

```diff
@@ -136,6 +136,11 @@ extern int s2n_config_add_ticket_crypto_key(struct s2n_config *config,
                                             uint8_t *key, uint32_t key_len,
                                             uint64_t intro_time_in_seconds_from_epoch);
 
+struct s2n_connection;
+
+typedef int s2n_keylog_callback_fn(struct s2n_connection *conn, const char *line, void *ctx);
```

Comment:
what is `ctx` used for? it doesn't seem needed. openssl doesn't have this option either. https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_keylog_callback.html

Reply:
Allows the user to pass in a pointer to a structure containing application specific data without resorting to hacks, required e.g. to reference the `FILE*` where the log shall reside.

That pattern is used throughout s2n for callbacks.

Reply:
I see, I initially thought that `s2n_connection` would be the context. (seems like the additional ctx is also sometimes referred as data in the other s2n functions using this pattern). Thanks for explaining.